backend/src/test/groovy/com/capstone/moneytree/controller/UserControllerTest.groovy

```diff
@@ -199,4 +201,54 @@ class UserControllerTest extends Specification {
         User user3 = createUser("joe@test.com", "Joe", "hello3", "Joe", "Wert", null)
         return Lists.asList(user1, user2, user3)
     }
+
+    @Test
+    def "Login Test"(){
+        given: "A registeredUser"
+        String password = "password123"
+        User registeredUser = User.builder()
```

Comment:
Whats the purpose of createUser? It seems to only be a wrapper for the user builder, but less flexible than the builder. I think that method doesn't add any purpose and adds clutter, while also being ambiguous as with the builder we know exactly what we are setting (e.g. email) for each strings.

Reply:
I had in mind that later when we'll have much more tests than right now, we would create a Util class for tests having all our builder creators inside. That is the purpose of createUser where instead of using User.builder() every time, we use createUser with the params inside. It's the same process: if you do not specify firstName in the builder for example, then it will be null anyway. I know that the method would become: createUser(email, username, null, null, null, null) but I feel like it hides the builder that we don't have to write every time we create a user here. I don't think builders should show up in tests and they should be hidden.

Reply:
We agree that in the back its the same process, but I'm still trying to understand this: 

> I feel like it hides the builder that we don't have to write every time we create a user here. I don't think builders should show up in tests and they should be hidden.

I personally find the builder to be more informative when reading the code and it removes the clutter of having an extra method. Why should we hide/wrap the builder?

Reply:
Creating a user or any entity should not be part of the test. We should have a static class Util called "CreateEntities" that creates all the needed builders for our tests. So in that case, we would have a common place to create our builders. In Spring, we can also do the same for Beans in our configuration if we wanted to. I know that both of them have the same process but it's just good **refactoring** in the long run in my opinion. It's always nice to have a strong testing base with static classes that are used to build objects used in tests.

Reply:
At work we have a *Utils file to create testing fixtures (users, objects...). We use the builder pattern for really big objects but for small objects, we use a utilitty method like setUpUser("Razine", "BEns, "Raz) and thats it. Both are okay, but it definitely is a good refactoring practice to encapsulate building the barebone users into a method if it is repeated across many tests.

We do not need it now, we may in the future, but still, it remains trivial.

Just my 2cents lol

Reply:
Thanks guys, I get it now. That util class does seem like a great idea. I'll use the createUser method for easier refactoring in the future.